common/mock_telemetry.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+
+	"github.com/stretchr/testify/mock"
+)
+
+type MockMetricHandle struct {
+	mock.Mock
+}
+
+func (m *MockMetricHandle) GCSReadBytesCount(ctx context.Context, inc int64) {
```

Comment:
why do we need mock for metricHandle. how are we testing so far?

Reply:
Current tests are composite tests(build on fake bucket) and not unit tests. For writing unit tests we have to mock metric handler behaviour.

e.g.  We have to mock [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/gcsx/random_reader.go#L254) line to write unit test for `tryReadingFromFileCache` function.